sktime/transformers/series_as_features/truncation.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import pandas as pd
+from sktime.transformers.series_as_features.base import \
+    BaseSeriesAsFeaturesTransformer
+from sktime.utils.validation.series_as_features import check_X
+
+__all__ = ["TruncationTransformer"]
+__author__ = ["Aaron Bostrom"]
+
+
+class TruncationTransformer(BaseSeriesAsFeaturesTransformer):
+    """TruncationTransformer docstring
+    """
+
+    def __init__(self, lower=None, upper=None):
+        self.lower = lower
+        self.upper = upper
+        self.min_length = lower
+        super(TruncationTransformer, self).__init__()
+
+    @staticmethod
+    def get_min_length(X):
+        def get_length(input):
+            return min(map(lambda series: len(series), input))
+
+        return min(map(get_length, X))
+
+    def fit(self, X, y=None):
+        """
+        Fit transformer.
+
+        Parameters
+        ----------
+        X : pandas DataFrame of shape [n_samples, n_features]
+            Input data
+        y : pandas Series, shape (n_samples, ...), optional
+            Targets for supervised learning.
+
+        Returns
+        -------
+        self : an instance of self.
+        """
+        X = check_X(X)
+
+        n_instances, _ = X.shape
+
+        arr = [X.iloc[i, :].values for i in range(n_instances)]
+
+        if self.lower is None:
+            self.lower = TruncationTransformer.get_min_length(arr)
+
+        self._is_fitted = True
+        return self
+
+    def transform(self, X, y=None):
+        """
+        Transform X.
+
+        Parameters
+        ----------
+        X : nested pandas DataFrame of shape [n_instances, n_columns]
+            Nested dataframe with time-series in cells.
+
+        Returns
+        -------
+        Xt : pandas DataFrame
+        """
+        self.check_is_fitted()
+        X = check_X(X)
+
+        n_instances, _ = X.shape
+
+        arr = [X.iloc[i, :].values for i in range(n_instances)]
+
+        min_length = TruncationTransformer.get_min_length(arr)
+
+        if min_length < self.lower:
+            raise ValueError(
+                "Error: min_length of series \
+                    is less than the one found when fit or set.")
+
+        # depending on inputs either find the shortest truncation.
+        # or use the bounds.
+        if self.upper is None:
+            idxs = np.arange(self.lower)
+        else:
+            idxs = np.arange(self.lower, self.upper)
+
+        truncate = [pd.Series([series[idxs]
```

Comment:
that should probably be `series.iloc[idxs]` when working with numerical indices, otherwise you'll look up the actual index which could be anything (e.g. time stamps)